browser-test/src/support/applicant_file_question.ts

```diff
@@ -39,6 +40,14 @@ export class ApplicantFileQuestion {
     expect(await this.page.innerHTML('body')).toContain(fileName)
   }
 
+  async expectFileNameCount(fileName: string, count: number) {
+    expect(
+      (await this.page.locator(this.uploadedFilesLocator).innerHTML()).split(
```

Comment:
I don't think that works because I need to await on the innerHTML promise<string>, so that I can get the string and split it and count the number of files. I'm guessing the other pattern works in some cases (where we're expect()ing a promise), but doesn't work in cases like this. But I'm very very far from an expert on this.

Reply:
@gwendolyngoetz would be the expert here so I'd go with what she says.

Reply:
You should make the _#cf-fileupload-uploaded-files_ a `<ul>` element with an aria-label. For this example I'm calling the aria-label "Upload Files", but call it whatever makes sense.

```typescript
await expect(
    this.page.getByRole('list', { name: 'Uploaded Files')
             .locator('li')
             .filter({ hasText: fileName})
)
.toHaveCount(count)
```

I think this will work, but it may need tweaking.

This way is preferred by Playwright. They call them [web first assertions](https://playwright.dev/docs/best-practices#use-web-first-assertions) and Playwright can better manage the waiting and watching and timeouts.

This example also uses `getByRole` which means we are testing the output int he same manner accessibility tools would be interacting with the site.